Sources/RandomAccessCollection+Sweet.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+extension CollectionType {
+    func enumerateWithNext() -> [(Any, Any?)] {
```

Comment:
Problem is: how do I define a `CollectionType` constrain that allows for `Element` to be an optional?